src/plugins/kernel_install/grub.sh

```diff
@@ -4,7 +4,29 @@
 # 1. run_bootloader_update: Update GRUB in a local and remote machine.
 # 2. run_bootloader_for_vm: Update GRUB in a virtual machine.
 
-declare -rg DEFAULT_GRUB_CMD_UPDATE='grub-mkconfig -o /boot/grub/grub.cfg'
+declare -g DEFAULT_GRUB_CMD_UPDATE='grub-mkconfig -o /boot/grub/grub.cfg'
+
+# Some distributions, such as Fedora, use GRUB2 as the default bootloader. On
+# those systems, grub-mkconfig command is replaced by grub2-mkconfig. This function
+# checks if the grub-mkconfig command exists and if doesn't, the default grub
+# update command is set to grub2-mkconfig.
+#
+# Returns:
+# 0 if a grub update command exists and 2 otherwise.
+function define_grub_cmd_update()
+{
+  local grub_cmd='grub-mkconfig'
+  local grub2_cmd='grub2-mkconfig'
+
+  if [[ ! -x "$(command -v $grub_cmd)" ]]; then
+    if [[ ! -x "$(command -v $grub2_cmd)" ]]; then
+      return 2 # ENOENT
```

Comment:
Could you add a test that hit this case?

Reply:
Added!